fabric-template-generator/src/index.ts

```diff
@@ -0,0 +1,92 @@
+import { addGradleWrapper } from './gradlewrapper';
+import { addGroovyGradle } from './gradlegroovy';
+import { getKotlinAdapterVersions, getLoaderVersions, getMinecraftYarnVersions } from './api';
+import { addModJson } from './modjson';
+
+export * as api from "./api";
+
+export interface Options {
+	/**
+	 * The configuration of the project we'd like to generate.
+	 */
+	config: Configuration;
+	/**
+	 * An interface we can write the generated files to, depending on the platform we're running
+	 * this might be directly to the filesystem or to a zip file.
+	 */
+	writer: TemplateWriter,
+	/**
+	 * Because the Gradle wrapper is a huge file we don't want to bundle it in our JavaScript so
+	 * we need to load it somehow. Depending on the platform how we load it will vary, so we'll
+	 * leave it up to the user to give us the file.
```

Comment:
I expect there may be other files like this. I did try to bundle it into the javascript but was unable to get it to work. Base64 encoding it first may be an option. May switch this out at a later date 👍